rest-api/etl/etl.sql

```diff
@@ -767,58 +716,47 @@ CREATE TABLE cdm.src_person_location
     zip varchar(255),
     state_ppi_code varchar(255),
     state varchar(255),
+    location_id bigint,
     PRIMARY KEY (participant_id)
 );
 
-DROP VIEW IF EXISTS cdm.temp_src_person_location;
-CREATE VIEW cdm.temp_src_person_location AS
-SELECT
-    src_m.participant_id,
-    src_m.question_ppi_code,
-    src_m.value_string,
-    IF(src_m.question_ppi_code = 'PIIAddress_StreetAddress',
-                src_m.value_string, NULL)                          AS address_1,
-    IF(src_m.question_ppi_code = 'PIIAddress_StreetAddress2',
-                src_m.value_string, NULL)                          AS address_2,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIICity',
-                src_m.value_string, NULL)                          AS city,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIIZIP',
-                src_m.value_string, NULL)                          AS zip,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIIState' AND src_m.topic_value = 'States',
-                src_m.value_ppi_code, NULL)                        AS state_ppi_code
-FROM cdm.src_mapped src_m
-INNER JOIN cdm.src_participant src_p
-    ON  src_m.participant_id = src_p.participant_id
-    AND src_m.date_of_survey = src_p.latest_date_of_location
-WHERE
-    src_m.question_ppi_code IN (
-        'PIIAddress_StreetAddress',
-        'PIIAddress_StreetAddress2',
-        'StreetAddress_PIICity',
-        'StreetAddress_PIIZIP',
-        'StreetAddress_PIIState'
-    )
-;
-
 INSERT cdm.src_person_location
 SELECT
-    src.participant_id                          AS participant_id,
-    MAX(src.address_1)                          AS address_1,
-    MAX(src.address_2)                          AS address_2,
-    MAX(src.city)                               AS city,
-    MAX(src.zip)                                AS zip,
-    MAX(src.state_ppi_code)                     AS state_ppi_code,
-    MAX(RIGHT(src.state_ppi_code, 2))           AS state
-FROM cdm.temp_src_person_location src
-GROUP BY
-    src.participant_id
-HAVING
-    COUNT(DISTINCT src.address_1) <= 2
```

Comment:
I don't see this constraint being applied after your changes, though I could easily have missed it.

Reply:
Yeah, this is a functional change. I'm checking these changes with the Odysseus team this morning to make sure they look to them.

I think the old logic may have been trying to only give an address to a person if they had only one associated address in our PPI data... but then these should have been < 2 instead of <= 2.

Instead I'm trying to change things to take the last address answer they gave.

Reply:
Odysseus confirmed these changes looked good.